### PR TITLE
Add HTML lexer

### PR DESCRIPTION
**Pull request by fabon-f**

### CODE REVIEW

File: src/noir/lexers/html.cr [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    rule /'/, Str, :pop!
+    rule /[^']+/, Str
+  end
+
+  state :script_content do
+    rule %r([^<]+) do |m|
+      m.delegate m.lexer.as(HTML).js_lexer
+    end
+
+    rule %r(<\s*/\s*script\s*>)m, Name::Tag, :pop!
```
Comment (makenowjust) [inline src/noir/lexers/html.cr]: It's a little question: why `m` flag is needed here? `m` flag does not affect this regexp.

**Review by makenowjust — COMMENTED** [review]
LGTM. However I left a little question. Could you answer (or fix) it?

File: src/noir/lexers/html.cr (hunk 1 above)
Comment (fabon-f) [reply src/noir/lexers/html.cr]: I think it is unnecessary. I'll remove it.

**Comment by makenowjust** [comment]
@fabon-f Thank you!

**Pull request merged**